### DISCUSSION TITLE
Does the library support ESP32 RTOS tasks?

### DISCUSSION THREAD

**Discussion by cdg123** [Bodmer/TFT_eSPI#954] [Q&A - General] [answered]
I am failry new to working with ESP32 and RTOS but aware having multiple tasks writing to the tft screen at once is a potential issue. At the moment I have it working ok but getting a corrupt screen at random times so wanted to check with you first if there was a best practice or example of how to manage this ? The library is working great apart from that thanks.

**Comment by Bodmer** [comment, marked as answer]
The library is not thread safe. One user posted that it works when run as a single task, but this is not something I have tried so am unable to provide help. You may need to migrate away from Arduino TFT libraries to find a thread safe one.

**Reply by cdg123** [reply]
Thanks for getting back so quickly. I had a dig into the FreeRTOS documents and have added a global semaphore handle (mutex)_(SemaphoreHandle_t xSemaphore4tft;)_ and then within each of the tasks that are running added an _if( xSemaphoreTake( xSemaphore4tft, ( TickType_t ) 100 ) == pdTRUE )_ before the tft activityand an  _xSemaphoreGive( xSemaphore4tft )_ immediately after which has been working  flawlessly. Thanks.
